observability/observability.go

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package observability implements the tracing, metrics, and logging data
+// collection, and provides controlling knobs via a config file.
+//
+// Experimental
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a
+// later release.
+package observability
+
+import (
+	"context"
+
+	"google.golang.org/grpc/binarylog"
+	"google.golang.org/grpc/grpclog"
+	configpb "google.golang.org/grpc/observability/internal/config"
+)
+
+var (
+	logger = grpclog.Component("observability")
+	config *configpb.ObservabilityConfig
+)
+
+func init() {
+	internalInit()
+}
+
+// internalInit exists to allow unit tests to re-parse ENV var.
+func internalInit() {
+	config = parseObservabilityConfig()
+
+	// Logging is controlled by the config at methods level. Users might bring
+	// their in-house exporter. If logging is disabled, binary logging also
+	// won't start. The overhead should be minimum.
+	startLogging(config)
```

Comment:
Why are we doing `startLogging` at init time, instead of waiting for `Start`?

I think this is poorly named?  `installLogger` or something?

Reply:
Yep... this is poorly named. Changed to `prepareLogging`.